Microsoft.Research/Contracts/MsCorlib/System.Threading.Tasks.Task.cs

```diff
@@ -1545,7 +1574,162 @@ public static Task<TResult> Run<TResult>(Func<TResult> function, CancellationTok
       return null;
 
     }
-    
+    //
+    // Summary:
+    //     Creates a System.Threading.Tasks.Task`1 that's completed successfully with the
+    //     specified result.
+    //
+    // Parameters:
+    //   result:
+    //     The result to store into the completed task.
+    //
+    // Type parameters:
+    //   TResult:
+    //     The type of the result returned by the task.
+    //
+    // Returns:
+    //     The successfully completed task.
+    [Pure]
+    public static Task<TResult> FromResult<TResult>(TResult result)
+    {
+      Contract.Ensures(Contract.Result<Task<TResult>>() != null);
+      Contract.Ensures(Contract.Result<Task<TResult>>().Exception == null);
+      Contract.Ensures(!Contract.Result<Task<TResult>>().IsCanceled);
+      Contract.Ensures(Contract.Result<Task<TResult>>().IsCompleted);
+      Contract.Ensures(!Contract.Result<Task<TResult>>().IsFaulted);
+      Contract.Ensures(Contract.Result<Task<TResult>>().Status == TaskStatus.RanToCompletion);
```

Comment:
Just curious would it be enough to leave just one postcondition that `Status == TaskStatus.RanToCompletion` and than express (via postconditions) relationship between `Status` and `IsCanceled`, `IsCompleted` and `IsFaulted`?

I'm fine with this approach but proposed one could be slightly easier to read, because all those postconditions:

```
+      Contract.Ensures(Contract.Result<Task<TResult>>().Exception == null);
 +      Contract.Ensures(!Contract.Result<Task<TResult>>().IsCanceled);
 +      Contract.Ensures(Contract.Result<Task<TResult>>().IsCompleted);
 +      Contract.Ensures(!Contract.Result<Task<TResult>>().IsFaulted);
```

Are just redundant.